include/sisl/cache/simple_cache.hpp

```diff
@@ -42,14 +43,27 @@ class SimpleCache {
             m_key_extract_cb{std::move(extract_cb)},
             m_map{num_buckets, m_key_extract_cb, std::bind(&SimpleCache< K, V >::on_hash_operation, this, _1, _2, _3)},
             m_per_value_size{per_val_size} {
-        m_record_family_id = m_evictor->register_record_family(std::move(evict_cb));
+        m_record_family_id = m_evictor->register_record_family(std::move(evict_cb), [this](const CacheRecord& record) {
+            V const value = reinterpret_cast<SingleEntryHashNode< V >*>(const_cast< CacheRecord* >(&record))->m_value;
+            K key = m_key_extract_cb(value);
+            t_evicted_keys.insert(key);
```

Comment:
what is the reason we want the removal delayed to insert time?

Reply:
Additional info, the post_eviction_cb is called by the hash_map while holding its lock